src/app/WriteHandler.cpp

```diff
@@ -486,7 +486,7 @@ FabricIndex WriteHandler::GetAccessingFabricIndex() const
     }
     else
     {
-        fabric = mpExchangeCtx->GetSessionHandle()->AsSecureSession()->GetFabricIndex();
+        fabric = mpExchangeCtx->GetSessionHandle()->GetFabricIndex();
```

Comment:
Again, please remove the now-unnecessary branching here.  This whole function should be:
```
  return mpExchangeCtx->GetSessionHandle()->GetFabricIndex();
```